ci-operator/step-registry/openshift-qe/hypershift-infra/openshift-qe-hypershift-infra-commands.sh

```diff
@@ -0,0 +1,140 @@
+        
+        log "$(date) - Restart stateful set pods"
+        echo "rollout restart -n openshift-monitoring statefulset/prometheus-k8s"
+        oc rollout restart -n openshift-monitoring statefulset/prometheus-k8s 
+        
+        log "$(date) - Wait till they are completely restarted"
+        oc rollout status -n openshift-monitoring statefulset/prometheus-k8s
+        
+        log "$(date) - Check pods status again and the hosting nodes"
+        oc get pods -n openshift-monitoring -o wide | grep prometheus-k8s
+    else
+        log "$(date) - Initiate migration of ingress router-default pods to infra nodepools"
+        echo "Add toleration to use infra nodes"
+
+        oc patch ingresscontroller -n openshift-ingress-operator default --type merge --patch  '{"spec":{"nodePlacement":{"nodeSelector":{"matchLabels":{"node-role.kubernetes.io/infra":""}},"tolerations":[{"effect":"NoSchedule","key":"node-role.kubernetes.io/infra","operator":"Exists"}]}}}'
+        
+        echo "Wait till it gets rolled out"
+        sleep 60
+
+        oc get pods -n openshift-ingress -o wide
+    fi    
+}
+
+function checkInfra() {
+    TRY=0
+    while [ $TRY -le 3 ]; do # Attempts three times to migrate pods
+        FLAG_ERROR=""
+        rebalanceInfra $1
+        for node in $(oc get pods -n "$2" -o wide | grep -i "$1" | grep -i running | awk '{print$7}');
+        do
+            if [[ $(oc get nodes --no-headers -l node-role.kubernetes.io/infra | awk '{print$1}' | grep $node) != "" ]]; then
+                log "$(date) - $node is an infra node"
+            else
+                log "$(date) - $1 pod on $node is not an infra node, retrying"
+                FLAG_ERROR=true
+            fi
+        done
+        if [[ $FLAG_ERROR == "" ]]; then return 0; else TRY=$((TRY+1)); fi
+    done
+    echo "Failed to move $1 pods in $2 namespace"
+    exit 1
+}
+
+# Get cluster 
+CLUSTER_ID=$(cat "${SHARED_DIR}/cluster-id")
+echo "CLUSTER_ID is $CLUSTER_ID"
+
+# Configure aws
+CLOUD_PROVIDER_REGION=${LEASED_RESOURCE}
+AWSCRED="${CLUSTER_PROFILE_DIR}/.awscred"
+if [[ -f "${AWSCRED}" ]]; then
+  export AWS_SHARED_CREDENTIALS_FILE="${AWSCRED}"
+  export AWS_DEFAULT_REGION="${CLOUD_PROVIDER_REGION}"
+else
+  echo "Did not find compatible cloud provider cluster_profile"
+  exit 1
+fi
+
+# Log in
+ROSA_VERSION=$(rosa version)
+ROSA_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token")
+if [[ ! -z "${ROSA_TOKEN}" ]]; then
+  echo "Logging into ${ROSA_LOGIN_ENV} with offline token using rosa cli ${ROSA_VERSION}"
+  rosa login --env "${ROSA_LOGIN_ENV}" --token "${ROSA_TOKEN}"
+  if [ $? -ne 0 ]; then
+    echo "Login failed"
+    exit 1
+  fi
+else
+  echo "Cannot login! You need to specify the offline token ROSA_TOKEN!"
+  exit 1
+fi
+
+# Check if this is a HCP cluster
+is_hcp_cluster="$(rosa describe cluster -c "$CLUSTER_ID" -o json  | jq -r ".hypershift.enabled")"
+log "hypershift.enabled is set to $is_hcp_cluster"
+
+if [[ "$is_hcp_cluster" == "true" ]]; then
```

Comment:
@mukrishn Do we need to add any check or friendly message and exit if the cluster is not hcp?

Reply:
This is separate chain, not included on any classic ROSA workflow but I will include a safe exit with a message.